topdown/cancel.go

```diff
@@ -0,0 +1,33 @@
+// Copyright 2017 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package topdown
+
+import (
+	"sync/atomic"
+)
+
+// Cancel defines the interface for cancelling topdown queries. Cancel
+// operations are thread-safe and idempotent.
+type Cancel interface {
+	Cancel()
+	Cancelled() bool
+}
+
+type cancel struct {
```

Comment:
The atomic store/load operations require *int32 but we can't use a type conversion from *cancel to *int32.